File: Python/Working with Numbers/51. Finding Number of times x digit occurs in a given input.py
number = int(input("Enter the Number :"))
digit = int(input("Enter the Digit to count :"))
count = 0
originalNumber = number

while (number > 0):
    if( number % 10 == digit ):
        count += 1
    number //= 10

print("The", digit, "ocuured in the", originalNumber, "for", count, "times")

# Simple Method

# number = input("Enter the Number :")
# digit = input("Enter the Digit to count :")
# count = number.count(digit)
# print(count)

File: Java/Working with Numbers/51. Finding Number of times x digit occurs in a given input.java
import java.util.Scanner;

class Main{
    public static void main(String [] args) {
        Scanner userInput = new Scanner(System.in);
        
        System.out.println("Enter the Number:");
        int number = userInput.nextInt();
        
        System.out.println("Enter the digit to Count :");
        int digit = userInput.nextInt();
        
        int currentNumber = number;
        int count = 0;
        if ( currentNumber == 0 && digit == 0 ) {
            count = 1;
        }
        else {
            while ( currentNumber > 0 ) {
                int lastDigit = currentNumber % 10;
                if( lastDigit == digit ) {
                    count++;
                }
                currentNumber /= 10;
            }
        }

        // Simple Method

        // System.out.println("Enter the Number:");
        // String number = userInput.next();
        
        // System.out.println("Enter the digit to Count :");
        // String digit = userInput.next();

        // int count = number.length()-number.replace(digit,"").length();
        // System.out.println(count);
    System.out.println("The "+ digit + " ocuured in the "+ number +" for "+ count+" times");
    userInput.close();
    }
}